rust/main/agents/relayer/src/test_utils/mock_base_builder.rs

```diff
@@ -0,0 +1,147 @@
+use std::{
+    collections::VecDeque,
+    sync::{Arc, Mutex},
+};
+
+use hyperlane_base::{settings::CheckpointSyncerBuildError, MultisigCheckpointSyncer};
+use hyperlane_core::{
+    accumulator::merkle::Proof, AggregationIsm, CcipReadIsm, Checkpoint, HyperlaneDomain,
+    HyperlaneMessage, InterchainSecurityModule, MultisigIsm, RoutingIsm, H256,
+};
+
+use crate::msg::metadata::{BaseMetadataBuilderTrait, IsmAwareAppContextClassifier};
+
+type ResponseList<T> = Arc<Mutex<VecDeque<T>>>;
+
+#[derive(Debug, Default)]
+pub struct MockBaseMetadataBuilderResponses {
```

Comment:
Kind of verbose mocking, but I think its relatively straightforward to understand and use.
Not sure if mockall has something similar.